### PR TITLE
remove a bit of dead code

### PR DESCRIPTION
**Pull request by moxian**
#### Summary
None

#### Purpose of change

I was wondering how do zipper bags slow the rate of spoilage, and came across this `item_pocket::process` overload that was seemingly doing it wrong. Turns out it is never actually invoked, so to the void it goes.

#### Describe the solution

Select code, press delete

#### Describe alternatives you've considered

None

#### Testing

The binary builds and launches successfully. Tests compile.

### CODE REVIEW

**Comment by Maleclypse** [comment]
It never worked and nothing is attempting to call it in json or it never worked and there is json that targets it despite it having never worked?

**Comment by moxian** [comment]
This is a C++ function, an overloaded one even (see [second definition](https://github.com/CleverRaven/Cataclysm-DDA/blob/248e05815f19b1b8fd3ca4a70335aec0361db882/src/item_pocket.cpp#L1982)). Json has no way to target it unless it's piped through other C++ code that does call this one. Unless, of course, we do direct function calls from json by symbol name , but even then JSON has no way to guess the mangled name of the resulting function.

Theer's absolutely no way JSON can break with this. And if it does, then I will be beyond shocked by the level of cursed code in this codebase.

**Comment by Maleclypse** [comment]
Let me rephrase this.  Is this code extraneous because all rotting items are processed as active items and don't need this process_items function that was going to catch some potential other item processing method?

**Comment by kevingranade** [comment]
I think this was just a mistake, not something intended to do something and never used for that purpose.
For good or ill the status quo is to process items in a player inventory we crawl the entire inventory. This doesn't quite have the scaling issues of doing the same across the whole map, so it's fine,  probably. 

Putting this another way,  this is so old that even if we did want to use it for something it would need every aspect scrutinized, easier to just start over.
